### PR TITLE
design-docs: introduce rolling stock collections

### PR DESCRIPTION
**Pull request by multun**

### CODE REVIEW

**Review by leovalais — APPROVED** [review]
LGTM as is, maybe a few endpoints could be useful as implementation guidelines. 

We could introduce the possibility of locking a collection upfront as this feature is likely to be requested down the line I believe.

**Comment by multun** [comment]
There will be no need for locking once permissions are implemented: an infra is locked if nobody has writer or greater rights

**Comment by leovalais** [comment]
> There will be no need for locking once permissions are implemented: an infra is locked if nobody has writer or greater rights

Ofc, you're right.

**Comment by multun** [comment]
I've added a sample migration sql. I trust the implementer with coming up with the crud for this.

**Review by SergeCroise — CHANGES_REQUESTED** [review]
@multun , please review my suggestion

**Comment by multun** [comment]
The suggestion is good but I can't apply it without closing this PR: I don't have push access to the repository anymore. I see two possible options:
- push to the branch yourself, and resolve your own comment
- I could fork, close this PR and open a new one

File: content/docs/reference/design-docs/rolling-stock-collections.en.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+## Proposal
+
+A concept of rolling stock collection would be introduced:
+- each rolling stock must belong to exactly one collection
+- rolling stocks do not have individual permissions; permissions are managed at the collection level
+- when creating a scenario, rolling stock collections need to be added
+
+Rolling stock collections have:
+- a name
```
Comment (woshilapin) [inline content/docs/reference/design-docs/rolling-stock-collections.en.md:28]: And probably an identifier? (but based on the `SQL` below, I guess the answer to my answer is yes)

File: content/docs/reference/design-docs/rolling-stock-collections.en.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- a description
+
+
+```sql
+create table rolling_stock_collection(
+  id  bigserial generated always as identity primary key,
+  name text not null,
```
Comment (woshilapin) [inline content/docs/reference/design-docs/rolling-stock-collections.en.md:35]:
```sql
  description text not null,
```

File: content/docs/reference/design-docs/rolling-stock-collections.en.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+);
+
+alter table rolling_stock
+add column collection bigserial references rolling_stock_collection on delete cascade not null;
+
+-- speed up cascade
+create index on rolling_stock_collection(created_by);
+create index on rolling_stock(collection);
+```
+
```
Comment (woshilapin) [inline content/docs/reference/design-docs/rolling-stock-collections.en.md:47]: I guess we also would like to add a `rolling_stock_collection_id` to the `timetable` table, isn't it?

**Comment by flomonster** [comment]
With the new permission model we've designed, it won't be necessary to have rolling stock groups.

**Pull request closed without merging**